client-tools/Shell/RequestFuzzer.cpp

```diff
@@ -0,0 +1,273 @@
+    CharOperation charOp = static_cast<CharOperation>(
+        _randContext.mt() % CharOperation::kMaxCharOpValue);
+    switch (charOp) {
+      case CharOperation::kAddString:
+        generateRandAlphaNumericString(input);
+        break;
+      case CharOperation::kAddInt32: {
+        input.append(std::to_string(generateRandInt32()));
+        break;
+      }
+      default:
+        TRI_ASSERT(false);
+    }
+  }
+}
+
+void RequestFuzzer::randomizeLineOperation(uint32_t numIts) {
+  uint32_t its = numIts;
+  while (its--) {
+    LineOperation lineOp = static_cast<LineOperation>(
+        _randContext.mt() % LineOperation::kMaxLineOpValue);
+    switch (lineOp) {
+      case LineOperation::kInjectRandByteInLine: {
+        if (_headerSplitInLines.size() > 1) {
+          uint32_t randPos = generateRandNumWithinRange<uint32_t>(
+              1, _headerSplitInLines.size() - 1);
+          if (generateRandNumWithinRange<uint32_t>(0, 99) < 10) {
+            generateRandAsciiChar(_headerSplitInLines[randPos]);
+          } else {
+            generateRandAlphaNumericChar(_headerSplitInLines[randPos]);
+          }
+        }
+        break;
+      }
+      case LineOperation::kCopyLine: {
+        if (_headerSplitInLines.size() > 5) {
+          uint32_t index = generateRandNumWithinRange<uint32_t>(
+              1, _headerSplitInLines.size() - 1);
+          _headerSplitInLines.emplace_back(_headerSplitInLines.at(index));
+        }
+        break;
+      }
+      case LineOperation::kAddLine: {
+        if (_usedKeys.size() <= _wordListForKeys.size() - 5) {
+          std::string keyName;
+          std::string value;
+          do {
+            uint32_t keyPos = generateRandNumWithinRange<uint32_t>(
+                0, _wordListForKeys.size() - 1);
+            if (_wordListForKeys[keyPos] != "random") {
+              keyName = _wordListForKeys[keyPos];
+            } else {
+              keyName.clear();
+              randomizeCharOperation(keyName, 1);
+            }
+          } while (!_usedKeys.emplace(keyName).second);
+
+          randomizeCharOperation(value, 1);
+          _headerSplitInLines.emplace_back(std::move(keyName) + ":" +
+                                           std::move(value));
+        }
+        break;
+      }
+      default:
+        TRI_ASSERT(false);
+    }
+  }
+}
+
+std::unique_ptr<fuerte::Request> RequestFuzzer::createRequest() {
+  _headerSplitInLines.clear();
+  auto req = std::make_unique<fuerte::Request>();
+  std::string header;
+  generateHeader(header);
+  if (_randReqType != rt::GET && _randReqType != rt::HEAD &&
+      _randReqType != rt::OPTIONS) {
+    velocypack::Builder builder;
+    generateBody(builder);
+    std::string bodyAsStr = builder.slice().toString();
+    header.append("Content-length:" + std::to_string(bodyAsStr.size()) +
+                  "\r\n");
+    req->addBinary(reinterpret_cast<uint8_t const*>(&bodyAsStr[0]),
+                   bodyAsStr.length());
```

Comment:
Suggestion: make it send JSON in 50% of cases, VPack in the other 50%.

Reply:
This probably requires us to set the "Content-type: x-application/velocypack"...
```
    header.append("Content-length:" + std::to_string(builder.slice().byteSize()) +
                  "\r\n");
    req->addBinary(builder.slice().start(), builder.slice(), byteSize());
```

Reply:
Addressed